File: C++17/백준/Bronze/11050. 이항 계수 1/이항 계수 1.cc
#include <iostream>
using namespace std;

long long factorial (int num) {
    if (num <= 1) return 1;
    return num * factorial (num - 1);
}

int main () {
    int n, k;
    cin >> n >> k;
    
    long long n_fac = factorial(n);
    long long k_fac = factorial(k);
    long long total_fac = factorial(n - k);
    
    long long result = n_fac / (k_fac * total_fac);
    
    cout << result;
    
    return 0;
}

File: Java/백준/Bronze/11050. 이항 계수 1/이항 계수 1.java
import java.util.Scanner;

public class Main {
    public static void main (String[] args) {
        Scanner sc = new Scanner (System.in);
        
        int n = sc.nextInt();
        int k = sc.nextInt();
        
        long n_fac = factorial(n);
        long k_fac = factorial(k);
        long total_fac = factorial(n - k);
        
        long result = n_fac / (k_fac * total_fac);
        
        System.out.println(result);
        
        sc.close();
    }
    
    public static long factorial (int num) {
        if (num <= 1) return 1;
        return num * factorial (num - 1);
    }
}